code-challenges.js

```diff
@@ -4,38 +4,143 @@
+// BUT UNFORTUNALTRLY I CAN T SORT IT. I TAKE IT BACK!!!!Q!! I CAN SORT!!!!!!!!!!!!!!
+// BECAUSE I HAVE SPENT LIKE 10 HOURS ON THIS PROBLEM AND THE SHOW MUST GO ON
+
+
+
+var fullArr1 = [4, 9, 0, "7", 8, true, "hey", 7, 199, -9, false, "hola"];
 // Expected output: [-9, 9, 7, 199]
-var fullArr2 = ["hello", 7, 23, -823, false, 78, null, "67", 6, "number"]
+
+fullArr1.sort();
+
+function filterArr(fullArr1) {
+    var onlyNums = [];
+    for (var i = 0; i < fullArr1.length; i++) {
+        if (typeof fullArr1[i] === "number" && fullArr1[i] % 2 === 1 | fullArr1[i] % 2 === -1) {
+            onlyNums.push(fullArr1[i]);
+        }
+    }
+     
+    return onlyNums;
+}
+console.log(filterArr(fullArr1));
+
+
+var fullArr2 = ["hello", 7, 23, -823, false, 78, null, "67", 6, "number"];
 // Expected output: [-823, 7, 23]
 
+fullArr2.sort();
+
+function filterArr2(fullArr2) {
+    var onlyNums2 = [];
+    for (var i = 0; i < fullArr2.length; i++) {
+        if (typeof fullArr2[i] === "number" && fullArr2[i] % 2 === 1 || fullArr2[i] % 2 === -1) {
+            onlyNums2.push(fullArr2[i]);
+        }
+    }
+     
+    return onlyNums2;
+}
+console.log(filterArr2(fullArr2));
 
 
 
 // --------------------3) Create a function that takes in a string of a single word and returns the middle letter of the word. If the word is an even number of letters, return the two middle letters.
 
-var middleLetters1 = "hello"
+var middleLetters1 = "hello";
 // Expected output: “l”
-var middleLetters2 = "rhinoceros"
+var middleLetters2 = "rhinoceros";
 // Expected output: “oc”
 
 
+// I AM A LITTLE RUSTY ON A FUNCTION HERE
+
+
+// NOW LETS PICK OUT THE MIDDLE
+
+function middle(arr){
+        if (middleLetters1[i] % 2 === 1){
+            return middleLetters1[3];
+        } 
+        if (middleLetters2[i] % 2 === 0){
+            return middleLetters2[4, 5];
+        }
+    }
 
 
-// --------------------4) Create a class to get the area of a sphere. Create three spheres with different radi as test cases. Area of a sphere =  4πr^2 (four pi r squared)
+console.log(middleLetters1[3]);
+console.log(middleLetters2[4], middleLetters2[5])
 
 
+// // --------------------4) Create a class to get the area of a sphere. Create three spheres with different radi as test cases. Area of a sphere =  4πr^2 (four pi r squared)
+// // I DONT KNOW WHAT I AM DOING IN THIS EXERCISE
+// import React, { Component } from 'react'
 
+// class Sphere extends Component {
+//   constructor(){
+//     super()
+//       this.state = {
+//         sides: 4
+//         pi: Math.PI
+//         rad: num * num
+//     }
+//   }
+
+//   handleChange = () => {
+//     let newShere1 = this.state.sides * this.state.pi * this.state.rad
+//     this.setState({ newShere1: newRad })
+//   }
+
+//   render() {
+//     let { rad } = this.state
+
+//     return (
+//       <div>
+//         <h2>Sphere: { rad }</h2>
+//       </div>
+//     )
+//   }
+// }
+
+// export default Sphere
```

Comment:
good approach but not quite the answer we are looking for. remember we can make a class on its own, it doesn't have to live in react.